runtime/manifest.js

```diff
@@ -367,8 +367,26 @@ ${e.message}
     for (let connection of items.connections) {
       let fromParticle = manifest.findParticleByName(connection.from.particle);
       let toParticle = manifest.findParticleByName(connection.to.particle);
-      assert(fromParticle, `could not find particle ${fromParticle}`);
-      assert(toParticle, `could not find particle ${toParticle}`);
+      if (!fromParticle) {
+        let error = new Error(`could not find particle '${connection.from.particle}'`);
```

Comment:
It looks to be old but I dig the literate `connection.from.particle` style code.

Reply:
the previous error wasn't working, as the fromParticle would be undefined when this error happens.